src/requests/features.jl

```diff
@@ -300,14 +300,23 @@ function _binding_kind(b, server)
     end
 end
 
-function julia_getModuleAt_request(params::TextDocumentPositionParams, server::LanguageServerInstance, conn)
-    doc = getdocument(server, URI2(params.textDocument.uri))
-    offset = get_offset(doc, params.position)
-    x = get_expr(getcst(doc), offset)
-    if x isa EXPR
-        scope = StaticLint.retrieve_scope(x)
-        if scope !== nothing
-            return get_module_of(scope)
+function julia_getModuleAt_request(params::VersionedTextDocumentPositionParams, server::LanguageServerInstance, conn)
+    uri = URI2(params.textDocument.uri)
+
+    if hasdocument(server, uri)
+        doc = getdocument(server, uri)
+        if doc._version == params.version
+            offset = get_offset(doc, params.position)
+            x = get_expr(getcst(doc), offset)
+            if x isa EXPR
+                scope = StaticLint.retrieve_scope(x)
+                if scope !== nothing
+                    return get_module_of(scope)
+                end
+            end
+        else
+            @warn "version mismatch in getModuleAt for $(uri): internal $(doc._version), in vscode: $(params.version)"
+            return nothing
```

Comment:
We can't return a `nothing` here, that is not a valid return value according to the request definition (I should actually add a check to JSONRPC.jl that throws an error if a handler does this kind of thing).

If you want to return an error condition, just return a `JSONRPCError` (don't throw it, just return it).

We should also definitely make sure we record this situation in the telemetry, just to understand whether version sync story _is_ actually the root case of this problem or not. The brute force version for now would be to define another one of our custom error types and throw an error here and crash for now, or you could also send a trace.